### PR TITLE
fix: only close server when server exists

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -220,7 +220,7 @@ class ClusterServer extends Base {
     // close server if no one is listening on this port any more
     if (!listening) {
       const server = serverMap.get(port);
-      yield server && server.close();
+      if (server) yield server.close();
```
Comment (fengmk2) [inline lib/server.js]: 要不然会抛 yield undefined 异常

image

Comment (gxcsoccer) [inline lib/server.js]: 额 +1

**Comment by gxcsoccer** [comment]
windows 一直不稳定，等后面重构的时候统一修复

**Comment by gxcsoccer** [comment]
\+ cluster-client@1.6.7

**Pull request merged**